### DISCUSSION TITLE
How get user IP when using a 'Proxy'

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pallets/flask#5027] [Q&A] [answered]
HI everyone,

I'm using nginx -> Gunicorn -> Flask

Great, but when I'm doing a 
```python
request.environ['REMOTE_ADDR']
```
I get the localhost (127.0.0.1)

So I suppose I have to setup nginx and Gunicorn (also ?) to forward the header ?

Thanks

**Comment by RonnyPfannschmidt** [comment, marked as answer]
by applying https://werkzeug.palletsprojects.com/en/2.2.x/middleware/proxy_fix/

this is intentionally not the default to protect apps from being passed those headers with malicious intent

**Reply by RonnyPfannschmidt** [reply]
whops, see https://flask.palletsprojects.com/en/2.2.x/deploying/proxy_fix/ for the exact fix

**Reply by BobSquarePants** [reply]
Thank you @RonnyPfannschmidt ,

So in this scenario `nginx - > Gunicorn -> Flask` 

how many number of proxies do I have 1 or 2 I guess 2.. ?

```python
from werkzeug.middleware.proxy_fix import ProxyFix

app.wsgi_app = ProxyFix(
    app.wsgi_app, x_for=2, x_proto=2, x_host=2, x_prefix=2
)
```

**Reply by RonnyPfannschmidt** [reply]
Typically gunicorn is not a proxy, but the wsgi server

My initial guess would be 1

You may need to validate proxy pass over in nginx


In case you run on something like kubernetes, there may be a additional ingres/routing proxy

**Reply by BobSquarePants** [reply]
It worked ! Thanks @RonnyPfannschmidt
